src/frontends/common/src/frontend.cpp

```diff
@@ -42,14 +43,16 @@ bool FrontEnd::supported_impl(const std::vector<ov::Any>& variants) const {
 InputModel::Ptr FrontEnd::load_impl(const std::vector<ov::Any>& variants) const {
     FRONT_END_CHECK_IMPLEMENTED(m_actual, load_impl);
     auto model = std::make_shared<InputModel>();
-    model->m_shared_object = m_shared_object;
+    if (m_shared_objects.size() > 0) {
+        model->m_shared_object = m_shared_objects[0];  // frontend plugin shared handle
```

Comment:
This code assumes that developer has 'private knowledge' that first item in a vector is a 'frontend plugin handle', not extension handle. Is it really better than having just one shared_ptr<void> which hides everything inside?